slick/src/main/scala/slick/jdbc/OracleProfile.scala

```diff
@@ -536,6 +537,24 @@ END;
       }).asInstanceOf[ResultConverter[JdbcResultConverterDomain, Option[T]]]
     else super.createOptionResultConverter(ti, idx)
 
+
+  private trait OverrideInsertAll[U] { self : InsertActionComposerImpl[U] =>
+    override def insertAll(values: Iterable[U], option: StatementOption = StatementOption.Batch): FixedSqlAction[MultiInsertResult, NoStream, Effect.Write] = {
+      option match {
+        case StatementOption.Batch =>
+          new self.MultiInsertAction(compiled.standardInsert, values)
+        case StatementOption.SingleStatement =>
+          throw new SlickException("OracleProfile doesn't support multiple insert with single statement.")
```

Comment:
I don't like runtime errors.

Is there any way to design it that only the right databases would have this API?

If you don't get to this by Sunday I'll try to work on that then.

Reply:
I thought that using macros might help (I haven't tried it at all), but since Scala3 support is coming up, I didn't adopt that idea this time.

Reply:
No the solution is to have more precise types. Sorry I didn't get to it yet

Reply:
I've implemented this as https://github.com/slick/slick/commit/7e60b4369e1c6aa2be3ab80c4e4331ded37465de -- thoughts?

I would do it as a separate PR.